File: examples/scala/test-coll-driver.scala
object TestCollDriver extends App
{
    val m = new M_TINY("Tiny");
    val t_Tiny = m.t_Result;
    val p = new TinyParser(m);

    val w = if (args.length > 0) {
      p.parseFile(args(0))
    } else {
      t_Tiny.f_root(t_Tiny.f_branch(t_Tiny.f_leaf(3),t_Tiny.f_leaf(4)))
    };

    if (args.contains("--debug")) Debug.activate();

    m.finish();

    val m2 = new M_TEST_COLL("Test Coll",m);
    val w2 = w.asInstanceOf[m2.T_Root];

    m2.finish();

    println("Results:");
    println("sum is " + m2.v_sum);
    println("leaves is " + m2.v_leaves);
    println("result is " + m2.v_result(w2));
}

File: examples/cpp/test-coll-driver.cpp
#include <iostream>
#include <stdexcept>
#include "aps-impl.h"
#include "basic.h"
#include "test-coll.h"

using namespace std;

typedef C_BAG<C_Integer> C_Integers;

int main()
{
  try {
    C_TINY m;

    C_SEQUENCE<C_TINY::C_Wood> seq_type(m.t_Wood);
    
    C_TINY::T_Wood w = m.v_branch(m.v_leaf(3),m.v_leaf(4));
    C_TINY::T_Root r = m.v_root(w);

    Debug::out(cout);

    m.finish();

    C_TEST_COLL<C_TINY::C_Result> m2(&m);

    m2.finish();

    cout << "sum is " << m2.v_sum << endl;
    cout << "leaves is "
	 << COLL<C_Integers,C_Integer>
            (m2.t_Integers,t_Integer).to_string(m2.v_leaves)
	 << endl;
    cout << "result is " << m2.v_result(r) << endl;
  } catch (exception& e) {
    cout << "Got error: " << e.what() << endl;
  }
}
